### PR TITLE
Libbpf sync 2023-06-13T17:50:39.365Z

### PR DESCRIPTION
**Pull request by chantra**
```
Dumping existing libbpf commit signatures...
WORKDIR:          /home/chantra/local/libbpf/libbpf
LINUX REPO:       /home/chantra/local/bpf
LIBBPF REPO:      /home/chantra/local/libbpf/libbpf
TEMP DIR:         /tmp/tmp.n4oUTpqfGC
SUFFIX:           2023-06-13T17-50-39.365Z
BASE COMMIT:      'fcf1fa29c8ea ("Merge branch 'libbpf: capability for resizing datasec maps'")'
TIP COMMIT:       '25085b4e9251 ("bpf/docs: Update documentation for new cpumask kfuncs")'
BPF BASE COMMIT:  'f726e03564ef ("bpf, sockmap: Test progs verifier error with latest clang")'
BPF TIP COMMIT:   'ad96f1c9138e ("bpf: Fix a bpf_jit_dump issue for x86_64 with sysctl bpf_jit_enable.")'
SQUASH COMMIT:    cfb944ef46474ff6aaf953fbc85ac8ea6de4acb4
BASELINE TAG:     libbpf-baseline-2023-06-13T17-50-39.365Z
TIP TAG:          libbpf-tip-2023-06-13T17-50-39.365Z
BPF BASELINE TAG: libbpf-bpf-baseline-2023-06-13T17-50-39.365Z
BPF TIP TAG:      libbpf-bpf-tip-2023-06-13T17-50-39.365Z
SQUASH BASE TAG:  libbpf-squash-base-2023-06-13T17-50-39.365Z
SQUASH TIP TAG:   libbpf-squash-tip-2023-06-13T17-50-39.365Z
VIEW TAG:         libbpf-view-2023-06-13T17-50-39.365Z
LIBBPF SYNC TAG:  libbpf-sync-2023-06-13T17-50-39.365Z
PATCHES:          /tmp/tmp.n4oUTpqfGC/patches
Picking '4c857a719bf9 ("libbpf: Change var type in datasec resize func")'...
Picking '59842c5451fe ("libbpf: Ensure libbpf always opens files with O_CLOEXEC")'...
Picking '4aadd2920b81 ("libbpf: Ensure FD >= 3 during bpf_map__reuse_fd()")'...
Picking '8ad77e72caae ("bpf: Add table ID to bpf_fib_lookup BPF helper")'...
Picking '132328e8e851 ("bpf: netfilter: Add BPF_NETFILTER bpf_attach_type")'...

Rewrite cfb944ef46474ff6aaf953fbc85ac8ea6de4acb4 (1/6) (0 seconds passed, remaining 0 predicted)    
Rewrite def8d6c0a67a9bc959e9407d0a79593dc7af1a60 (2/6) (20 seconds passed, remaining 40 predicted)    
Rewrite e9789a8ef940cdecd64bf2047378eabac3a965e6 (3/6) (22 seconds passed, remaining 22 predicted)    
Rewrite 665987d7bd6c719411a43ce0f60d31d2e675a84d (4/6) (25 seconds passed, remaining 12 predicted)    
Rewrite 595ef654583af5209c563b0c2e08c4503e493fdb (5/6) (27 seconds passed, remaining 5 predicted)    
Rewrite 59f496fc2e9638fd14b77bde65af2710e22ff0ec (6/6) (29 seconds passed, remaining 0 predicted)    
Ref 'refs/heads/libbpf-squash-tip-2023-06-13T17-50-39.365Z' was rewritten
Ref 'refs/heads/libbpf-squash-base-2023-06-13T17-50-39.365Z' was rewritten

Rewrite 6062632d8a60a4ccf69f267316c5a48ae9a76043 (1/6) (0 seconds passed, remaining 0 predicted)    
Rewrite e60d22130719f0abfcf1848e5566db44af59307e (2/6) (0 seconds passed, remaining 0 predicted)    
Rewrite 65740405434f7996fe86874ad88ec13b9768ae10 (3/6) (0 seconds passed, remaining 0 predicted)    
Rewrite 6335dc30f92568fb5b55fc56b536ba0645ddace8 (4/6) (0 seconds passed, remaining 0 predicted)    
Rewrite 51e41b43dcbfa700f80107deee11e2aaa5a95278 (5/6) (0 seconds passed, remaining 0 predicted)    
Rewrite 6662a0b6ede678525d4cad57719a7ad75667e7ba (6/6) (0 seconds passed, remaining 0 predicted)    
Ref 'refs/heads/libbpf-squash-tip-2023-06-13T17-50-39.365Z' was rewritten
Ref 'refs/heads/libbpf-squash-base-2023-06-13T17-50-39.365Z' was rewritten
/tmp/tmp.n4oUTpqfGC/patches/0000-cover-letter.patch
/tmp/tmp.n4oUTpqfGC/patches/0001-libbpf-Change-var-type-in-datasec-resize-func.patch
/tmp/tmp.n4oUTpqfGC/patches/0002-libbpf-Ensure-libbpf-always-opens-files-with-O_CLOEX.patch
/tmp/tmp.n4oUTpqfGC/patches/0003-libbpf-Ensure-FD-3-during-bpf_map__reuse_fd.patch
/tmp/tmp.n4oUTpqfGC/patches/0004-bpf-Add-table-ID-to-bpf_fib_lookup-BPF-helper.patch
/tmp/tmp.n4oUTpqfGC/patches/0005-bpf-netfilter-Add-BPF_NETFILTER-bpf_attach_type.patch
Applying: libbpf: Change var type in datasec resize func
Using index info to reconstruct a base tree...
M	src/libbpf.c
Falling back to patching base and 3-way merge...
No changes -- Patch already applied.
Applying: libbpf: Ensure libbpf always opens files with O_CLOEXEC
Applying: libbpf: Ensure FD >= 3 during bpf_map__reuse_fd()
Applying: bpf: Add table ID to bpf_fib_lookup BPF helper
Applying: bpf: netfilter: Add BPF_NETFILTER bpf_attach_type
[libbpf-sync-2023-06-13T17-50-39.365Z 06fd3b6] sync: auto-generate latest BPF helpers
 1 file changed, 4 insertions(+)
[libbpf-sync-2023-06-13T17-50-39.365Z b919fff] sync: latest libbpf changes from kernel
 2 files changed, 2 insertions(+), 2 deletions(-)
SUCCESS! 5 commits synced.
Verifying Linux's and Github's libbpf state

Rewrite 25085b4e9251c77758964a8e8651338972353642 (1/1) (0 seconds passed, remaining 0 predicted)    
Ref 'refs/heads/libbpf-view-2023-06-13T17-50-39.365Z' was rewritten

Rewrite 3f88245ffceafa5287dc84c32536e5450c76d2bc (1/1) (0 seconds passed, remaining 0 predicted)    
Ref 'refs/heads/libbpf-view-2023-06-13T17-50-39.365Z' was rewritten
Comparing list of files...
Comparing file contents...
--- /home/chantra/local/bpf/include/uapi/linux/bpf.h	2023-06-13 10:51:51.036073937 -0700
+++ /home/chantra/local/libbpf/libbpf/include/uapi/linux/bpf.h	2023-06-13 10:51:35.754091412 -0700
@@ -1035,6 +1035,7 @@
 	BPF_TRACE_KPROBE_MULTI,
 	BPF_LSM_CGROUP,
 	BPF_STRUCT_OPS,
+	BPF_NETFILTER,
 	__MAX_BPF_ATTACH_TYPE
 };
 
/home/chantra/local/bpf/include/uapi/linux/bpf.h and /home/chantra/local/libbpf/libbpf/include/uapi/linux/bpf.h are different!
--- /home/chantra/local/bpf/src/libbpf.c	2023-06-13 10:52:22.368038114 -0700
+++ /home/chantra/local/libbpf/libbpf/src/libbpf.c	2023-06-13 10:51:35.761091404 -0700
@@ -117,6 +117,7 @@
 	[BPF_PERF_EVENT]		= "perf_event",
 	[BPF_TRACE_KPROBE_MULTI]	= "trace_kprobe_multi",
 	[BPF_STRUCT_OPS]		= "struct_ops",
+	[BPF_NETFILTER]			= "netfilter",
 };
 
 static const char * const link_type_name[] = {
@@ -8736,7 +8737,7 @@
 	SEC_DEF("struct_ops+",		STRUCT_OPS, 0, SEC_NONE),
 	SEC_DEF("struct_ops.s+",	STRUCT_OPS, 0, SEC_SLEEPABLE),
 	SEC_DEF("sk_lookup",		SK_LOOKUP, BPF_SK_LOOKUP, SEC_ATTACHABLE),
-	SEC_DEF("netfilter",		NETFILTER, 0, SEC_NONE),
+	SEC_DEF("netfilter",		NETFILTER, BPF_NETFILTER, SEC_NONE),
 };
 
 static size_t custom_sec_def_cnt;
/home/chantra/local/bpf/src/libbpf.c and /home/chantra/local/libbpf/libbpf/src/libbpf.c are different!
--- /home/chantra/local/bpf/src/libbpf_probes.c	2023-06-13 10:52:22.369038112 -0700
+++ /home/chantra/local/libbpf/libbpf/src/libbpf_probes.c	2023-06-13 10:51:35.770091394 -0700
@@ -180,7 +180,9 @@
 	case BPF_PROG_TYPE_SK_REUSEPORT:
 	case BPF_PROG_TYPE_FLOW_DISSECTOR:
 	case BPF_PROG_TYPE_CGROUP_SYSCTL:
+		break;
 	case BPF_PROG_TYPE_NETFILTER:
+		opts.expected_attach_type = BPF_NETFILTER;
 		break;
 	default:
 		return -EOPNOTSUPP;
/home/chantra/local/bpf/src/libbpf_probes.c and /home/chantra/local/libbpf/libbpf/src/libbpf_probes.c are different!
Unfortunately, there are some inconsistencies, please double check.
Ok, proceeding...
Cleaning up...
Your branch is up to date with 'bpf-next/master'.
Deleted branch libbpf-baseline-2023-06-13T17-50-39.365Z (was fcf1fa29c8ea).
Deleted branch libbpf-tip-2023-06-13T17-50-39.365Z (was 25085b4e9251).
Deleted branch libbpf-bpf-baseline-2023-06-13T17-50-39.365Z (was f726e03564ef).
Deleted branch libbpf-bpf-tip-2023-06-13T17-50-39.365Z (was ad96f1c9138e).
Deleted branch libbpf-squash-base-2023-06-13T17-50-39.365Z (was 94a0e7b4f8f3).
Deleted branch libbpf-squash-tip-2023-06-13T17-50-39.365Z (was 9a43ec9e1268).
Deleted branch libbpf-view-2023-06-13T17-50-39.365Z (was b508f5fa79fc).
DONE.
```

### CODE REVIEW

**Comment by chantra** [comment]
bpf_nf/tc-bpf-ct:FAIL
has been failing on both s390x and kernel LATEST test_progs_noalu32 since the last sync.
https://github.com/libbpf/libbpf/actions/runs/5085607205/jobs/9139233610 is the first time we see it so I would guess the issue started as we sync libbpf last time: https://github.com/libbpf/libbpf/pull/693

For some reason, it does not repro in kernel-patches/bpf, so not sure why that could be.

**Comment by chantra** [comment]
More precisely, the error looks like:
```
2023-06-13T19:40:12.8750649Z test_bpf_nf_ct:PASS:test_bpf_nf__open_and_load 0 nsec
2023-06-13T19:40:12.8762711Z test_bpf_nf_ct:PASS:iptables-legacy -t raw -A PREROUTING -j CONNMARK --set-mark 42/0 0 nsec
2023-06-13T19:40:12.8765256Z test_bpf_nf_ct:PASS:start_server 0 nsec
2023-06-13T19:40:12.8770061Z connect_to_server:PASS:socket 0 nsec
2023-06-13T19:40:12.8773499Z connect_to_server:PASS:connect_fd_to_fd 0 nsec
2023-06-13T19:40:12.8778436Z test_bpf_nf_ct:PASS:connect_to_server 0 nsec
2023-06-13T19:40:12.8781124Z test_bpf_nf_ct:PASS:accept 0 nsec
2023-06-13T19:40:12.8784197Z test_bpf_nf_ct:PASS:sockaddr len 0 nsec
2023-06-13T19:40:12.8789265Z test_bpf_nf_ct:PASS:bpf_prog_test_run 0 nsec
2023-06-13T19:40:12.8793558Z test_bpf_nf_ct:PASS:Test EINVAL for NULL bpf_tuple 0 nsec
2023-06-13T19:40:12.8800894Z test_bpf_nf_ct:PASS:Test EINVAL for reserved not set to 0 0 nsec
2023-06-13T19:40:12.8805290Z test_bpf_nf_ct:PASS:Test EINVAL for netns_id < -1 0 nsec
2023-06-13T19:40:12.8812339Z test_bpf_nf_ct:PASS:Test EINVAL for len__opts != NF_BPF_CT_OPTS_SZ 0 nsec
2023-06-13T19:40:12.8818727Z test_bpf_nf_ct:PASS:Test EPROTO for l4proto != TCP or UDP 0 nsec
2023-06-13T19:40:12.8823546Z test_bpf_nf_ct:PASS:Test ENONET for bad but valid netns_id 0 nsec
2023-06-13T19:40:12.8829498Z test_bpf_nf_ct:PASS:Test ENOENT for failed lookup 0 nsec
2023-06-13T19:40:12.8834334Z test_bpf_nf_ct:PASS:Test EAFNOSUPPORT for invalid len__tuple 0 nsec
2023-06-13T19:40:12.8840069Z test_bpf_nf_ct:PASS:Test for alloc new entry 0 nsec
2023-06-13T19:40:12.8843960Z test_bpf_nf_ct:PASS:Test for insert new entry 0 nsec
2023-06-13T19:40:12.8854643Z test_bpf_nf_ct:FAIL:Test for successful lookup unexpected Test for successful lookup: actual -2 != expected 0
2023-06-13T19:40:12.8865471Z test_bpf_nf_ct:FAIL:Test for min ct timeout update unexpected Test for min ct timeout update: actual 0 <= expected 8
2023-06-13T19:40:12.8871354Z test_bpf_nf_ct:PASS:Test for max ct timeout update 0 nsec
2023-06-13T19:40:12.8886114Z test_bpf_nf_ct:FAIL:Test for insert and lookup mark value unexpected Test for insert and lookup mark value: actual 0 != expected 77
2023-06-13T19:40:12.8891933Z test_bpf_nf_ct:FAIL:Test for ct status update  unexpected Test for ct status update : actual 0 != expected 442
2023-06-13T19:40:12.8897516Z test_bpf_nf_ct:PASS:Test existing connection lookup 0 nsec
2023-06-13T19:40:12.8902276Z test_bpf_nf_ct:PASS:Test existing connection lookup ctmark 0 nsec
2023-06-13T19:40:12.8913037Z test_bpf_nf_ct:FAIL:Test for source natting unexpected Test for source natting: actual -22 != expected 0
2023-06-13T19:40:12.8926826Z test_bpf_nf_ct:FAIL:Test for destination natting unexpected Test for destination natting: actual -22 != expected 0
2023-06-13T19:40:12.8928744Z #17/2    bpf_nf/tc-bpf-ct:FAIL
2023-06-13T19:40:12.8932871Z #17      bpf_nf:FAIL
2023-06-13T19:40:12.8942079Z Summary: 366/3139 PASSED, 29 SKIPPED, 1 FAILED
```

File: ci/vmtest/configs/DENYLIST-latest [hunk 1]
```diff
@@ -1 +1,2 @@
 decap_sanity  # weird failure with decap_sanity_ns netns already existing, TBD
+bpf_nf/tc-bpf-ct # test consistently failing on x86: https://github.com/libbpf/libbpf/pull/698#issuecomment-1590341200
```
Comment (anakryiko) [inline ci/vmtest/configs/DENYLIST-latest:2]: seems like we need to add `bpf_nf/xdp-ct` here as well?

**Review by anakryiko — CHANGES_REQUESTED** [review]
please use `ci: ` prefix for commits updating denylist, for consistent naming with other commits

**Review by anakryiko — APPROVED** [review]

**Pull request merged**